scripts/deleteAndLoadSnapsets.js

```diff
@@ -0,0 +1,327 @@
+    });
+    fluid.log("\tSnapset Prefs Safes marked for deletion.");
+    return dbLoader.snapsetPrefsSafes;
+};
+
+/**
+ * Find the GPII Key records that are associated with a snapset PrefsSafe, mark
+ * them for deletion, and add them to array of records to delete.
+ * @param {String} responseString - The response from the database query for
+ *                                  retrieving all the GPII Keys.
+ * @return {Array} - The GPII Key records marked for deletion.
+ */
+dbLoader.processGpiiKeys = function (responseString) {
+    fluid.log("Processing the GPII Keys...");
+    var gpiiKeyRecords = JSON.parse(responseString);
+    dbLoader.gpiiKeys = dbLoader.markPrefsSafesGpiiKeysForDeletion(
+        gpiiKeyRecords, dbLoader.snapsetPrefsSafes
+    );
+    fluid.log("\tGPII Keys associated with snapset Prefs Safes marked for deletion.");
+    return dbLoader.gpiiKeys;
+};
+
+/**
+ * Given all the GPII Keys records in the database, find the ones that reference
+ * a snapset PrefsSafe.  As each GPII Key is found it is marked for
+ * deletion.
+ * @param {Array} gpiiKeyRecords - Array of GPII Key records from the database.
+ * @param {Array} snapSets - Array of snapset Prefs Safes whose id references
+ *                           its associated GPII Key record.
+ * @return {Array} - the values from the gpiiKeyRecords that are snapset GPII Keys.
+ */
+dbLoader.markPrefsSafesGpiiKeysForDeletion = function (gpiiKeyRecords, snapSets) {
+    var gpiiKeysToDelete = [];
+    fluid.each(gpiiKeyRecords.rows, function (gpiiKeyRecord) {
+        var gpiiKey = fluid.find(snapSets, function (aSnapSet) {
+            if (gpiiKeyRecord.value.prefsSafeId === aSnapSet._id) {
+                return gpiiKeyRecord.value;
+            }
+        }, null);
+        if (gpiiKey !== null) {
+            gpiiKey._deleted = true;
+            gpiiKeysToDelete.push(gpiiKey);
+        }
+    });
+    return gpiiKeysToDelete;
+};
+
+/**
+ * Utility to wrap all the pieces to make a bulk documents deletion request
+ * for the snapset Prefs Safes and their associated GPII keys.  Its intended use
+ * is the parameter of the appropriate promise.then() call.
+ */
+dbLoader.doBatchDelete = function () {
+    var docsToRemove = dbLoader.snapsetPrefsSafes.concat(dbLoader.gpiiKeys);
+    var execBatchDelete = dbLoader.createBulkDocsRequest(
+        docsToRemove, dbLoader.batchDeleteResponse
+    );
+    execBatchDelete();
+};
+
+/**
+ * Create a function that makes a bulk docs POST request using the given data.
+ * @param {Object} dataToPost - JSON data to POST and process in bulk.
+ * @param {Object} responseHandler - http response handler for the request.
+ * @return {Function} - A function that wraps an http request to execute the
+ *                      POST.
+ */
+dbLoader.createBulkDocsRequest = function (dataToPost, responseHandler) {
+    return function () {
+        var postOptions = {
+            hostname: dbLoader.parsedCouchDbUrl.hostname,
```

Comment:
Please avoid fishing these variables out as globals and instead accept a structured options argument which encodes them

Reply:
Okay